exp/internal/controllers/machinepool_controller.go

```diff
@@ -57,6 +58,13 @@ import (
 const (
 	// MachinePoolControllerName defines the controller used when creating clients.
 	MachinePoolControllerName = "machinepool-controller"
+
+	// stateConfirmationTimeout is the amount of time allowed to wait for desired state.
+	stateConfirmationTimeout = 10 * time.Second
+
+	// stateConfirmationInterval is the amount of time between polling for the desired state.
+	// The polling is against a local memory cache.
+	stateConfirmationInterval = 100 * time.Millisecond
```

Comment:
nit: since those values are used only inside `waitForMachineCreation`, let's move this const inside the func.
also, possibly let's name them something like `waitForCacheUpdateTimeout/Interval` because IMO `stateConfirmationTimeout/Interval` is a little bit too generic.